e2e-test-server/e2e_test_server/scenarios.py

```diff
@@ -124,6 +135,36 @@ def basic_propagator(request: Request) -> Response:
     return Response(status_code=code_pb2.OK, headers={TRACE_ID: trace_id})
 
 
+def detect_resource(request: Request) -> Response:
+    """Create a trace with GCP resource detector"""
+
+    # Temporarily skip if on Cloud Run until it is implemented
+    if "K_CONFIGURATION" in os.environ:
+        return Response(
+            status_code=code_pb2.UNIMPLEMENTED,
+            data=b"Resource detection not yet implemented on Cloud Run",
+        )
+
+    with _tracer_setup(
+        tracer_provider_config={
+            "resource": GoogleCloudResourceDetector(
+                raise_on_error=True
+            ).detect()
+        },
+        exporter_config={"resource_regex": r".*"},
+    ) as tracer:
+        propagator = CloudTraceFormatPropagator()
```

Comment:
should we be using the XCTC propagator here?

Reply:
Oops copypasta mistake thanks.